bot.py

```diff
@@ -25,18 +29,34 @@ async def format_embed(self, name, anime=False):
         except kadal.MediaNotFound:
             return
 
-        title = (media.title.get('english')
-                 or media.title.get('romaji')
-                 or media.title.get('native'))
+        title = (media.title.get("english")
+                 or media.title.get("romaji")
+                 or media.title.get("native"))
         desc = "***" + ", ".join(media.genres) + "***\n"
         if media.description is not None:
             desc += media.description[:256 - len(desc)] + f"... [(more)]({media.site_url})"
         # dirty half-fix until i figure something better out
-        desc = desc.replace("<br>", "").replace("<i>", "").replace("</i>", "")
-        footer = re.sub(r'.*\.', '', str(media.format))
-        e = discord.Embed(title=title, description=desc, color=0x4286f4)
+        desc = desc.replace("<br>", "").replace("<bR>", "").replace("<Br>", "").replace("<BR>", "").replace("<i>", "").replace("</i>", "").replace("<I>", "").replace("</I>", "")
+        footer = re.sub(r".*\.", "", str(media.format))
+        title_cover_info = "https://img.anili.st/media/" + str(media.id) 
+
+        def title_cover_info_color(title_cover_info=title_cover_info,temp_file="tmp.jpg"):
+
+            opener = urllib.request.build_opener()
```

Comment:
urllib is not a good choice here for 2 reasons:
- It has a very barebones and sluggish API, which makes for ugly code and annoying boilerplate. A better use would be the `requests` library, which is much more modern and is widely used as the main HTTP library in python. Though, because of the second point it won't fit us here.
- It doesn't support asynchronous operation. This means every time it has to request the image, the whole bot will wait for it to finish and only then continue listening for messages to parse. This gets worse and worse the bigger the bot/server gets, and in particularly bad cases can straight up crash the bot.

consider using an async requests library with a modern API. Kadal, the library this bot uses to query AL (which I wrote) uses either `asks` or `aiohttp` depending on the environment. Consider looking into those.